Cesium.CodeGen/Ir/Expressions/FunctionCallExpression.cs

```diff
@@ -39,36 +39,78 @@ public IExpression Lower(IDeclarationScope scope)
         var functionName = _function.Identifier;
         var callee = scope.Functions.GetValueOrDefault(functionName)
                      ?? throw new CompilationException($"Function \"{functionName}\" was not found.");
+        int firstVarArgArgument = 0;
+        if (callee.Parameters is { } parameters)
+        {
+            if (parameters.IsVarArg)
+            {
+                firstVarArgArgument = parameters.Parameters.Count;
+            }
+        }
+
         return new FunctionCallExpression(
             _function,
             callee,
-            _arguments.Select(a => a.Lower(scope)).ToList());
+            _arguments.Select((a, index) =>
+            {
+                if (index >= firstVarArgArgument)
+                {
+                    var expressionType = a.GetExpressionType(scope);
+                    if (expressionType.Equals(scope.CTypeSystem.Float))
+                    {
+                        // Seems to be float always use float-point registers and as such we need to covert to double.
+                        return new TypeCastExpression(scope.CTypeSystem.Double, a.Lower(scope));
+                    }
+                    else
+                    {
+                        return a.Lower(scope);
+                    }
+                }
+
+                return a.Lower(scope);
+            }).ToList());
     }
 
     public void EmitTo(IEmitScope scope)
     {
+        VariableDefinition? varArgBuffer = null;
         var explicitParametersCount = _callee!.Parameters?.Parameters.Count ?? 0;
+        var varArgParametersCount = _arguments.Count - explicitParametersCount;
+        if (_callee!.Parameters?.IsVarArg == true)
+        {
+            // Using sparse population of the parameters on the stack. 8 bytes should be enough for anybody.
```

Comment:
_(Minor)_ Are vararg struct parameters allowed? Because, you know, we can create a struct of any size.

(we can merge this as-is of course, that's more of a discussion matter and not a complaint)

Reply:
Passing structs are possible. But I need to come up with lowering process which preserve information about expression type.  That simply need some work.

Reply:
At this point I almost like to have all expressions to have special field/property `ExpressionType` which is set by some specific compiler phase.

Reply:
Okay, let's at least add a TODO here. "8 bytes should be enough for anybody" sounds a little bit raw, especially if we know it isn't enough :)